CVAE_testbed/datasets/quilt_aics_features.py

```diff
@@ -0,0 +1,239 @@
+        
+        num_of_cells = len(non_categorical_dataframe)
+        print(num_of_cells)
+
+        # Here we normalize X with min max
+        # x = non_categorical_dataframe.values
+        # min_max_scaler = preprocessing.MinMaxScaler()
+        # x_scaled = min_max_scaler.fit_transform(x)
+        # non_categorical_dataframe = pd.DataFrame(x_scaled)
+
+        # This is mean, std normalization
+        non_categorical_dataframe = non_categorical_dataframe.iloc[:, 34:36]
+        # non_categorical_dataframe[(np.abs(stats.zscore(non_categorical_dataframe) < 3)).all(axis=1)]
+        # non_categorical_dataframe.apply(zscore)
+        print(non_categorical_dataframe.shape)
+        x = non_categorical_dataframe.values
+        # x_train_scaled = x[:34000, :]
+        # x_test_scaled = x[34000:, :]
+        std_scaler = preprocessing.StandardScaler()
+        x_train_scaled = std_scaler.fit_transform(x[:30000, :])
+        x_test_scaled = std_scaler.transform(x[30000:, :])
+
+        non_categorical_train = pd.DataFrame(x_train_scaled)
+        non_categorical_test = pd.DataFrame(x_test_scaled)
+        print(non_categorical_train.shape, non_categorical_test.shape )
+
+
+
+        # x_train_scaled = sklearn.preprocessing.normalize(x[:28000, 1:6], axis=0)
+        # x_test_scaled = sklearn.preprocessing.normalize(x[28000:, 1:6], axis=0)
+        # non_categorical_train = pd.DataFrame(x_train_scaled)
+        # non_categorical_test = pd.DataFrame(x_test_scaled)
+        # print(non_categorical_train.shape, non_categorical_test.shape )
+        # non_categorical_train = pd.DataFrame(np.concatenate((x[:28000, 0:1], x_train_scaled), axis = 1))
+        # non_categorical_test = pd.DataFrame(np.concatenate((x[28000:, 0:1], x_test_scaled), axis = 1))
+        
+        # non_categorical_dataframe = non_categorical_dataframe.drop_duplicates()
+        # non_categorical_dataframe = non_categorical_dataframe[(np.abs(stats.zscore(non_categorical_dataframe))<3).all(axis=1)]
+        
+        # Convert to torch tensor
+        all_non_categorical_X_train = torch.from_numpy(non_categorical_train.values).float()
+        all_non_categorical_X_test = torch.from_numpy(non_categorical_test.values).float()
+        all_categorical_X = torch.from_numpy(one_hot_categorical_features.values).float()
+
+
+        for j, i in enumerate(range(self.num_batches)):
+            X_train = all_non_categorical_X_train[i*self.BATCH_SIZE: (i+1)*self.BATCH_SIZE, :]
+            X_test = all_non_categorical_X_test[i*self.BATCH_SIZE: (i+1)*self.BATCH_SIZE, :]
+            X_categorical = all_categorical_X[i*self.BATCH_SIZE: (i+1)*self.BATCH_SIZE, :]
+
+            # X = torch.cat((X, X_categorical), 1)
+
+            if X_train.size()[0] != self.BATCH_SIZE:
+                break
+            print(X_train.size(), X_test.size(), Batches_X_train.size(), Batches_X_test.size())
```

Comment:
Again, prefer `logging` over `print` but I don't know how you have your `bin` script setup

Reply:
agree. will move to logging, print is habit